### PR TITLE
[Merged by Bors] - chore: update std4 to b197bd2, catching up to leanprover/std4#427

### PR DESCRIPTION
**Pull request by tobiasgrosser**
---

- [x] depends on: #8901
- [x] depends on: #8897
- [x] depends on: #8566

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
We merged https://github.com/leanprover-community/mathlib4/pull/8566 to see if we can get the update through.

File: Mathlib/MeasureTheory/Integral/Marginal.lean [hunk 1]
```diff
@@ -101,7 +101,14 @@ theorem _root_.Measurable.lmarginal (hf : Measurable f) : Measurable (∫⋯∫
...
 
-/-- The marginal distribution is independent of the variables in `s`. -/
+/-- The marginal distribution is independent of the variables in `s`.
+
+TODO: This fails with:
+
+elaboration function for 'Std.ExtendedBinder.«termSatisfies_binder_pred%__»'
+  has not been implemented
+  satisfies_binder_pred% i ∉ s
+
```
Comment (tobiasgrosser) [inline Mathlib/MeasureTheory/Integral/Marginal.lean:111]: This is the last remaining breakage. Help appreciated.

Comment (Ruben-VandeVelde) [reply Mathlib/MeasureTheory/Integral/Marginal.lean:111]: Uggh. Feel free to revert leanprover/std4#425

Comment (tobiasgrosser) [reply Mathlib/MeasureTheory/Integral/Marginal.lean:111]: The offending std4 PR is https://github.com/leanprover/std4/pull/425. Maybe @Ruben-VandeVelde knows why this broke.

**Comment by eric-wieser** [comment]
I think it would be better to update to a previoua version first

File: Mathlib/MeasureTheory/Integral/Marginal.lean (hunk 1 above)
Comment (tobiasgrosser) [reply Mathlib/MeasureTheory/Integral/Marginal.lean]: The revert has been committed to std4. This issues has been resolved.

**Comment by eric-wieser** [comment]
> > I think it would be better to update to a previous version first, so as not to also pull in #8566 at the same time.
> > I've done that in #8897
> 
> This will break the mathlib build as [leanprover/std4@c4ece13](https://github.com/leanprover/std4/commit/c4ece13ae2b82ce66d95bf865911ab68c3157315) is broken.

I think it will be fine, I pushed a trivial workaround

**Comment by eric-wieser** [comment]
In the interest of shrinking the patch, I think it might make sense to bump first to leanprover/std4#197, then to leanprover/std4#89.

**Comment by eric-wieser** [comment]
In fact, I think this ended up duplicating the work in #6202

**Comment by tobiasgrosser** [comment]
> In the interest of shrinking the patch, I think it might make sense to bump first to [leanprover/std4#197](https://github.com/leanprover/std4/pull/197), then to [leanprover/std4#89](https://github.com/leanprover/std4/pull/89).

I updated https://github.com/leanprover-community/mathlib4/pull/8566 to take care of [leanprover/std4#89](https://github.com/leanprover/std4/pull/89.

**Comment by eric-wieser** [comment]
@semorrison, I think that commit should have been pushed to #8566 instead; the two have diverged, and the other one is the one that addressed review comments. I've added the dependency to the description now.

**Comment by semorrison** [comment]
> @semorrison, I think that commit should have been pushed to #8566 instead; the two have diverged, and the other one is the one that addressed review comments. I've added the dependency to the description now.

Thanks. I've separately merged master into #8566 now, incorporating the same comments.

File: Mathlib/Data/List/Perm.lean [hunk 2]
```diff
@@ -230,7 +230,7 @@ lemma subperm_iff : l₁ <+~ l₂ ↔ ∃ l, l ~ l₂ ∧ l₁ <+ l := by
   exact ⟨l₁ ++ l', (h₂.trans (h₁.append_right _)).symm, (prefix_append _ _).sublist⟩
 
 -- This is now in `Std`, but apparently misnamed as `List.subperm_singleton_iff`.
```
Comment (eric-wieser) [inline Mathlib/Data/List/Perm.lean:232]: Can we prove the mathlib version in terms of the std one?

**Comment by semorrison** [comment]
I'm correcting the possible naming mistake in Std at https://github.com/leanprover/std4/pull/429. This PR doesn't need to wait for that; the existing comments will be enough breadcrumb to clean up when that PR causes a new name collision!

File: Mathlib/Data/List/Perm.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/Data/List/Perm.lean:232]: Ah sorry, wrong PR

**Comment by eric-wieser** [comment]
As @YaelDillies comments in #8897, there are a few more extended binders that it would be best to restore before merging.

**Comment by semorrison** [comment]
> As @YaelDillies comments in #8897, there are a few more extended binders that it would be best to restore before merging.

Done.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**